From 30d0a722a437eb21508f4ded90fa84ca4e464d49 Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 15:21:47 -0400
Subject: [infiniband] ocrdma: Memory leak fix in ocrdma_dereg_mr()

Message-id: <e3c235d74946bbde0466f5b795519bff49d5796b.1411571481.git.dledford@redhat.com>
Patchwork-id: 96614
O-Subject: [Patch RHEL7 12/48] RDMA/ocrdma: Memory leak fix in ocrdma_dereg_mr()
Bugzilla: 1081067 1088577
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Selvin Xavier <selvin.xavier@emulex.com>

Signed-off-by: Selvin Xavier <selvin.xavier@emulex.com>
Signed-off-by: Devesh Sharma <devesh.sharma@emulex.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 9d1878a369b23f48a5ca5bcbd89abb0e569c58cc)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c b/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
index d30747f..e1b373c 100644
--- a/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
+++ b/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
@@ -838,8 +838,7 @@ int ocrdma_dereg_mr(struct ib_mr *ib_mr)
 
  status = ocrdma_mbx_dealloc_lkey(dev, mr->hwmr.fr_mr, mr->hwmr.lkey);
 
- if (mr->hwmr.fr_mr == 0)
-  ocrdma_free_mr_pbl_tbl(dev, &mr->hwmr);
+ ocrdma_free_mr_pbl_tbl(dev, &mr->hwmr);
 
  /* it could be user registered memory. */
  if (mr->umem)
-- 
1.7.1